Src/Controllers/Frontend/WebhookController.php

```diff
@@ -0,0 +1,134 @@
+        $this->monduInvoice = new MonduInvoice();
+        $this->monduOrder = new MonduOrder();
+        $this->order = new Order();
+    }
+
+    /**
+     * @return array
+     */
+    public function handleWebhook()
+    {
+        $requestData = $this->request->all();
+
+        $params = [
+            'order_id' => $requestData['order_id'],
+            'order_number' => $requestData['order_uuid'],
+            'invoice_uuid' => $requestData['invoice_uuid'],
+            'order_state' => $requestData['order_state']
+        ];
+
+        switch ($requestData['topic']) {
+            case 'order/confirmed':
+            case 'order/canceled':
+            case 'order/declined':
+            case 'order/pending':
+                return $this->handleOrderStateChanged($params);
+            case 'invoice/created':
+                return $this->handleInvoiceStateChanged($params, 'created');
+            case 'invoice/canceled':
+                return $this->handleInvoiceStateChanged($params, 'canceled');
+            default:
+                return [['error' => 'Unregistered topic'], Response::HTTP_UNPROCESSABLE_ENTITY];
+        }
+    }
+
+    /**
+     * @param $params
+     *
+     * @return array
+     */
+    public function handleOrderStateChanged($params)
+    {
+        $monduOrder = $this->getOrder($params['order_number']);
+
+        if ($monduOrder) {
+            $monduOrder->update(['state' => $params['order_state']], $params['order_id']);
+
+            if ($params['order_state'] == 'pending') {
+                $this->updateOrderWithPendingStatus($monduOrder);
+            }
+
+            return [['order' => $monduOrder], Response::HTTP_OK];
+        }
+
+        return [['error' => 'Order not found'], Response::HTTP_BAD_REQUEST];
+    }
+
+    /**
+     * @param $params
+     * @param $state
+     *
+     * @return array
+     */
+    public function handleInvoiceStateChanged($params, $state)
+    {
+        $monduInvoice = $this->getInvoice($params['invoice_uuid']);
+
+        if ($monduInvoice) {
+            $monduInvoice->update(['state' => $state], $params['invoice_uuid']);
+            return [['invoice' => $monduInvoice], Response::HTTP_OK];
+        }
+
+        return [['error' => 'Invoice not found'], Response::HTTP_BAD_REQUEST];
+    }
+
+    /**
+     * @param $orderUuid
+     *
+     * @return mixed
+     */
+    private function getOrder($orderUuid)
+    {
+        return $this->monduOrder->select('order_uuid')->where('external_reference_id', $orderUuid)->first()[0];
+    }
+
+    /**
+     * @param $invoiceUuid
+     *
+     * @return mixed
+     */
+    private function getInvoice($invoiceUuid)
+    {
+        return $this->monduInvoice->select('invoice_uuid, order_id')->where('external_reference_id', $invoiceUuid)->first()[0];
+    }
+
+    /**
+     * @param $monduOrder
+     *
+     * @return int
+     */
+    private function updateOrderWithPendingStatus($monduOrder)
```

Comment:
Can we update the order status here? right now I think it's being set to paid. But for pending orders maybe we can use something else.

Reply:
@tikohov20 
https://guide.jtl-software.com/en/jtl-wawi/procurement/purchase-orders/#status-for-purchase-orders 

I wasn't able to find any other proper status and saw that in similar cases, we added a comment to the order, e.g.
https://github.com/mondu-ai/bnpl-checkout-woocommerce/blob/main/src/Mondu/Mondu/Controllers/WebhooksController.php#L108

Maybe you have better idea how we can handle this.